src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/common/GetFormatDataCommand.cs

```diff
@@ -169,27 +170,38 @@ protected override void ProcessRecord()
                 viewList.Add(formatdef);
             }// foreach(ViewDefinition...
 
-            // write out all the available type definitions
-            foreach (var pair in typedefs)
+            if (typedefs.Count == 0)
             {
-                var typeNames = pair.Key;
-
-                if (writeOldWay)
+                ErrorRecord errorRecord = new ErrorRecord(
+                    new TypeLoadException("No such type could be found"),
+                    "SPECIFIED_TYPE_NOT_FOUND",
```

Comment:
For the FullQualifiedErrorId, the convention is to use Pascal casing: SpecifiedTypeNotFound.

Reply:
Good point, @SteveL-MSFT.

Additionally, it should be a _non-terminating_ error, given that the cmdlet accepts multiple inputs and potentially produces multiple outputs.

@faraazahmad: I know I suggested `ResourceUnavailable`  initially, but on further reflection it probably isn't the right category.

We can use `Remove-TypeData` as a model, which already reports an error (also non-terminating), and uses category `InvalidOperation`:

```powershell
PS> Remove-TypeData NoSuchType
Remove-TypeData : Error in TypeData "NoSuchType": The type "NoSuchType" was not found. The type name value must be the full name of the type. Verify the type name and run the command again.
At line:1 char:1
+ Remove-TypeData NoSuchType
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [Remove-TypeData], RuntimeException
+ FullyQualifiedErrorId : TypesDynamicRemoveException,Microsoft.PowerShell.Commands.RemoveTypeDataCommand
```

As an aside: I'm not sure `TypesDynamicRemoveException` was a great choice of error ID.

@SteveL-MSFT, @iSazonov, what do you think?

Reply:
Yes, this should be a non-terminating error if multiple are accepted so no need to stop the entire operation.  Agree that `TypesDynamicRemoveException` is a bad error id, but it's a breaking change to change it now as the intent of the FQEI is to make it internet searchable.  For symmetry, it makes sense to use `InvalidOperation` and probably makes more sense for an end user whereas `TypeLoadException` is more dev-centric.